### PR TITLE
Progress

### LINKED ISSUE #12: Account api

### LINKED ISSUE #13: shop api

### PR DESCRIPTION
**Pull request by Vghxv**

### CODE REVIEW

File: model/account.py [hunk 1]
```diff
@@ -1,8 +1,20 @@
-from typing import Optional
-
+from typing import Optional, List
 from pydantic import BaseModel
-
 from utils.as_form import as_form
+from model.general import SuccessModel
+class Account(BaseModel):
```
Comment (kyynk) [inline model/account.py]: 空一行

**Review by kyynk — CHANGES_REQUESTED** [review]

File: model/shop.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel
+from typing import Optional, List
+from model.general import SuccessModel
+from utils.as_form import as_form
+
+class GetShopModel(BaseModel):
```
Comment (PEKKA1117) [inline model/shop.py]: Different  naming standard from Account.

File: model/shop.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    shop_uuid: str
+    account_uuid: str
+    name: str
+    description: str
+    image_url: str
+    update_time: str
+class ReturnShopModel(SuccessModel):
```
Comment (PEKKA1117) [inline model/shop.py]: Blank line missing.

File: router/account.py [hunk 4]
```diff
@@ -4,48 +4,129 @@
...
+                image_url,
+                email,
+                phone,
+                birthday,
+                address,
+                is_active,
+                update_time FROM `Account`
+                where account_uuid = %s;
+        """
+        result = get_all_result(sql, (account_uuid,))
```
Comment (PEKKA1117) [inline router/account.py]:
```suggestion
        sql = """
            SELECT
                account_uuid,
                name,
                image_url,
                email,
                phone,
                birthday,
                address,
                is_active,
                update_time FROM `Account`
        """
    if account_uuid:
        sql += " WHERE account_uuid = %s"
    sql += ";"
    result = get_all_result(sql, (account_uuid,))
```
would be clearer

File: router/account.py [hunk 5]
```diff
@@ -4,48 +4,129 @@
...
+        VALUES(
+            UUID(), %s, %s, %s, %s, %s, %s, %s, %s, DEFAULT, DEFAULT
+        );
+    """
+    result = execute_query(sql, tuple(account_form.values()))
+    if result:
+        return JSONResponse(
+            status_code=status.HTTP_200_OK,
+            content=jsonable_encoder({
+                "msg": "success"
```
Comment (PEKKA1117) [inline router/account.py]: return user_uuid here?

File: router/shop.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    elif account_uuid:
+        sql = """
+            SELECT * FROM `Shop` WHERE account_uuid = %s;
+        """
+        result = get_all_result(sql, (account_uuid,))
+    else:
+        sql = """
+            SELECT * FROM `Shop`;
+        """
+        result = get_all_result(sql)
```
Comment (PEKKA1117) [inline router/shop.py]:
```suggestion
  sql = "SELECT * FROM `Shop`"
    params = []

    if shop_uuid:
        sql += " WHERE shop_uuid = %s"
        params.append(shop_uuid)
    elif account_uuid:
        sql += " WHERE account_uuid = %s"
        params.append(account_uuid)

    result = get_all_result(sql, params if params else None)
 ```

would be clearer.

File: router/shop.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+})
+async def create_shop(
+    shop_form: CreateShopForm = Depends(CreateShopForm.as_form)
+):
+    shop_form = shop_form.model_dump()
+    sql = """
+        INSERT INTO `Shop`
+        VALUES (UUID(), %s, %s, %s, %s, DEFAULT);
+    """
+    result = execute_query(sql, tuple(shop_form.values()))
```
Comment (PEKKA1117) [inline router/shop.py]: It is NOT banning users from creating multi shops, which violates our data model.

**Review by Dragonfly911117 — CHANGES_REQUESTED** [review]

File: router/shop.py (hunk 7 above)
Comment (Vghxv) [reply router/shop.py]: db account_uuid should be unique in Shop table

Comment (PEKKA1117) [reply router/shop.py]:
```sql
alter table Shop
    add constraint Shop_pk
        unique (account_uuid);
```
something like this?

Comment (PEKKA1117) [reply router/shop.py]: Is this solved?

**Review by Dragonfly911117 — APPROVED** [review]
LGTM

**Pull request merged**